util/ranger/detacher.go

```diff
@@ -619,6 +710,9 @@ type DetachRangeResult struct {
 	AccessConds []expression.Expression
 	// RemainedConds is the filter conditions which should be kept after access.
 	RemainedConds []expression.Expression
+	// ColumnValues records the constant column values for all index columns.
+	// For the ith column, if it is evaluated as constant, ColumnValues[i] is its value. Otherwise ColumnValues[i] is nil.
+	ColumnValues []*valueInfo
 	// EqCondCount is the number of equal conditions extracted.
 	EqCondCount int
```

Comment:
This var can be deleted now?

Reply:
There is slight difference between how `ColumnValues` and `EqCondCount` are calculated. For example,
```
mysql> create table t2(a int, b int, c int, d int, index idx_a_b_c_d(a, b, c, d));
Query OK, 0 rows affected (0.01 sec)

mysql> explain select * from t2 where ((a = 1 and b = 1 and d < 3) or (a = 1 and b = 1 and d > 6)) and c = 3 order by d;
+---------------------------+---------+-----------+-----------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------+
| id                        | estRows | task      | access object                           | operator info                                                                                                                                        |
+---------------------------+---------+-----------+-----------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------+
| IndexReader_14            | 0.00    | root      |                                         | index:Selection_13                                                                                                                                   |
| └─Selection_13            | 0.00    | cop[tikv] |                                         | eq(test.t2.c, 3), or(and(eq(test.t2.a, 1), and(eq(test.t2.b, 1), lt(test.t2.d, 3))), and(eq(test.t2.a, 1), and(eq(test.t2.b, 1), gt(test.t2.d, 6)))) |
|   └─IndexRangeScan_12     | 10.00   | cop[tikv] | table:t2, index:idx_a_b_c_d(a, b, c, d) | range:[1,1], keep order:true, stats:pseudo                                                                                                           |
+---------------------------+---------+-----------+-----------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------+
```
In the case, `ColumnValues` detect `a, b, c` are constant columns while `EqCondCount` only detect `a` is a constant column. Do we need to improve it in this pr or unify `ColumnValues` and `EqCondCount` in another pr?

Reply:
I think we can improve it later, not in this PR. What's your opinion? @qw4990 @time-and-fate

Reply:
I think it's OK to improve it later.